[torch export] replace traceback.extract_stack with CapturedTraceback.extract

strisunshinewentingwang opened the pull request:
Summary:
with a simple bench in TestDeserializer.test_basic function: 
```
time_start = time.time()
for i in range(1000):
    self.check_graph(MyModule(), inputs)
warnings.warn(f"time_taken: {time.time() - time_start}")
```
and forcing FakeTensorConfig.debug to True, record_stack_traces to True, logging level to debug, it shows that the the changed code is consistently ard 20 secs faster (~90s vs originally ~110s)

Test Plan:
test passed, see summary

compared debug trace before and after: 
- exactly the same for fake tensor and proxy callsite https://www.internalfb.com/intern/diffing/?paste_number=1189883685
- slightly different for the user frame in proxy node https://www.internalfb.com/intern/diffing/?paste_number=1189884347

Differential Revision: D54237017




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang

strisunshinewentingwang commented:
> [<img alt="CLA Missing ID" width="328" height="28" src="https://camo.githubusercontent.com/83fc0261dbb20a7b16cc05d156131773ab142311e3e272338a23d3a26129bdc3/68747470733a2f2f73332e616d617a6f6e6177732e636f6d2f636c612d70726f6a6563742d6c6f676f2d70726f642f636c612d6d697373696e672d69642e737667">](https://api.easycla.lfx.linuxfoundation.org/v2/repository-provider/github/sign/29311126/65600975/121449/#/?version=2) [<img alt="CLA Not Signed" width="328" height="28" src="https://camo.githubusercontent.com/031843f73aecaf0b300b9aef2fc62377af0790eac726b513bf4ab20b714fbf8a/68747470733a2f2f73332e616d617a6f6e6177732e636f6d2f636c612d70726f6a6563742d6c6f676f2d70726f642f636c612d6e6f742d7369676e65642e737667">](https://api.easycla.lfx.linuxfoundation.org/v2/repository-provider/github/sign/29311126/65600975/121449/#/?version=2)
> 
> * ❌ The email address for the commit ([0f4fa77](https://github.com/pytorch/pytorch/commit/0f4fa77f4e8244d9a732050e89316882800f76a0)) is not linked to the GitHub account, preventing the EasyCLA check. Consult [this Help Article](https://confluence.linuxfoundation.org/pages/viewpage.action?pageId=86641302) and [GitHub Help](https://help.github.com/en/github/committing-changes-to-your-project/why-are-my-commits-linked-to-the-wrong-user) to resolve. (To view the commit's email address, add .patch at the end of this PR page's URL.) For further assistance with EasyCLA, [please submit a support request ticket](https://jira.linuxfoundation.org/servicedesk/customer/portal/4).

/easycla

angelayi reviewed (approved):
there's a linting error

strisunshinewentingwang committed:
[torch export] replace traceback.extract_stack with CapturedTraceback.extract (#121449)

Summary:

with a simple bench in TestDeserializer.test_basic function: 
```
time_start = time.time()
for i in range(1000):
    self.check_graph(MyModule(), inputs)
warnings.warn(f"time_taken: {time.time() - time_start}")
```
and forcing FakeTensorConfig.debug to True, record_stack_traces to True, logging level to debug, it shows that the the changed code is consistently ard 20 secs faster (~90s vs originally ~110s)

Test Plan:
test passed, see summary

compared debug trace before and after: 
- exactly the same for fake tensor and proxy callsite https://www.internalfb.com/intern/diffing/?paste_number=1189883685
- slightly different for the user frame in proxy node https://www.internalfb.com/intern/diffing/?paste_number=1189884347

Reviewed By: angelayi

Differential Revision: D54237017
